weblab/experiments/tests/test_views.py

```diff
@@ -520,41 +520,42 @@ def test_submatrix_with_too_many_model_ids(self, client, helpers, quick_experime
 
     def test_submatrix_with_protocol_versions(self, client, helpers, quick_experiment_version):
         exp = quick_experiment_version.experiment
-        v1 = str(exp.protocol_version)
-        v2 = str(helpers.add_fake_version(exp.protocol).sha)
+        v1 = exp.protocol_version
+        v2 = helpers.add_fake_version(exp.protocol)
         helpers.add_fake_version(exp.protocol)  # v3, not used
 
         exp2 = make_experiment(
-            exp.model, exp.model_version,
+            exp.model, exp.model.repocache.get_version(exp.model_version.sha),
             exp.protocol, v2,
         )
 
         # Add an experiment with a different protocol, which shouldn't appear
-        other_protocol = recipes.model.make()
+        other_protocol = recipes.protocol.make()
         other_protocol_version = helpers.add_fake_version(other_protocol, 'public')
-        make_experiment(exp.model, exp.model_version, other_protocol, other_protocol_version.sha)
+        make_experiment(exp.model, exp.model.repocache.get_version(exp.model_version.sha),
```

Comment:
Can you pass `exp.model_version` directly? Similarly on the next line?

Reply:
thanks done